### PR TITLE
ci: Migrate canaries to GitHub actions

### PR DESCRIPTION
**Pull request by dnys1**
- Adds canaries workflow to GitHub actions
- Consolidates Android emulator launching to shared action
- Consolidates iOS simulator launching to shared action
- Adds full e2e test for canaries on top of simple build test

### CODE REVIEW

**Comment by dnys1** [comment]
Example of passing workflow: https://github.com/aws-amplify/amplify-flutter/actions/runs/4801188871/jobs/8543092765?pr=2917 (3.3 tests should be failing currently).

File: build-support/build_canary.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# copy template files to newly generated app
+cp -r $ROOT_DIR/canaries/lib .
+cp $ROOT_DIR/.circleci/dummy_amplifyconfiguration.dart lib/amplifyconfiguration.dart
+
+# Android
+sed -i'' -e "s/minSdkVersion .*/minSdkVersion 24/" ./android/app/build.gradle && cat ./android/app/build.gradle
+# iOS
+sed -i'' -e "s/# platform .*/platform :ios, '13.0'/" ./ios/Podfile && cat ./ios/Podfile
+
+flutter build $@
```
Comment (ragingsquirrel3) [inline build-support/build_canary.sh:31]: Question: Am I correct in understanding this does not modify the main.dart file to add the plugins like the CCI canary does? It seems like that is tested with the e2e action, so trying to make sure I understand separation of concerns here, that this isn't testing that part on purpose?

File: .github/workflows/amplify_canaries.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      matrix:
+        flutter-version:
+          - "any" # latest
+          - "3.3.0"
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+        with:
+          persist-credentials: false
+
+      - name: Install dependencies
```
[the remark this answers is not in the document]
Comment (dnys1) [reply .github/workflows/amplify_canaries.yaml:26]: This runs the partial bs, which is necessary for the e2e tests (which don't use pubdev versions).

File: build-support/build_canary.sh (hunk 1 above)
Comment (dnys1) [reply build-support/build_canary.sh:31]: It copies over `lib/` above.

- e2e canary tests integration of all libraries in a quick test
- build canary verifies the latest pubdev packages can be built into a binary

**Review by ragingsquirrel3 — APPROVED** [review]

File: canaries/test/widget_test.dart
Comment (HuiSF) [inline canaries/test/widget_test.dart:1]: This file can be removed?

**Review by HuiSF — APPROVED** [review]

**Pull request merged**